jvb/src/main/kotlin/org/jitsi/videobridge/dcsctp/DcSctpTransport.kt

```diff
@@ -40,19 +42,51 @@ class DcSctpTransport(
     parentLogger: Logger
 ) {
     val logger = createChildLogger(parentLogger)
-    lateinit var socket: DcSctpSocketInterface
+    private val lock = Any()
+    private var socket: DcSctpSocketInterface? = null
 
     fun start(callbacks: DcSctpSocketCallbacks, options: DcSctpOptions = DEFAULT_SOCKET_OPTIONS) {
-        socket = factory.create(name, callbacks, null, options)
+        synchronized(lock) {
+            socket = factory.create(name, callbacks, null, options)
+        }
     }
 
     fun handleIncomingSctp(packetInfo: PacketInfo) {
         val packet = packetInfo.packet
-        socket.receivePacket(packet.getBuffer(), packet.getOffset(), packet.getLength())
+        synchronized(lock) {
+            socket?.receivePacket(packet.getBuffer(), packet.getOffset(), packet.getLength())
+        }
+    }
+
+    fun stop() {
+        synchronized(lock) {
+            socket?.close()
+            socket = null
+        }
+    }
+
+    fun connect() {
+        synchronized(lock) {
+            socket?.connect()
```

Comment:
Should this throw an error when `socket` is `null`?

Reply:
I don't think so - it just means that the socket was torn down before it could be connected, which should be fine.